cmd/duffle/relocate.go

```diff
@@ -97,100 +105,80 @@ duffle relocate helloworld path/to/relocatedbundle.json --repository-prefix exam
 }
 
 func (r *relocateCmd) run() error {
-	bun, err := r.setup()
+	rel, bun, tmpDir, err := r.setup()
 	if err != nil {
 		return err
 	}
+	defer os.RemoveAll(tmpDir)
 
-	if err := r.relocate(bun); err != nil {
+	if err := rel.Relocate(); err != nil {
 		return err
 	}
 
 	return r.writeBundle(bun)
 }
 
-func (r *relocateCmd) relocate(bun *bundle.Bundle) error {
-	// mutate the input bundle to become the output bundle
-	if !r.outputBundleIsFile {
-		bun.Name = r.outputBundle
+// The caller is responsible for deleting the returned temporary directory, which may contain the returned bundle.
+func (r *relocateCmd) setup() (*relocator.Relocator, *bundle.Bundle, string, error) {
```

Comment:
You could also return a "cleaning" functor, so the caller has only to defer the call of this function, without having to know what is the cleaning logic.

Reply:
Great idea. Done.

nit: it was merely a function, not a functor.